packages/pn-personafisica-webapp/src/pages/NotificationDetail.page.tsx

```diff
@@ -44,20 +44,40 @@ const NotificationDetail = () => {
   const { t } = useTranslation(['common', 'notifiche']);
   const isMobile = useIsMobile();
   const notification = useAppSelector((state: RootState) => state.notificationState.notification);
-  const currentRecipient = notification.recipients[0];
+
+
   /**
-   * REFERS TO: PN-1724
-   * The following code has been commented out and substituted with the line above
-   * due to issue PN-1724 since we currently do not have enough information to pick
-   * the right recipient assuming a multi-recipient notification, but this feature 
-   * is beyond the MVP scope. 
-   * 
-   const currentUser = useAppSelector((state: RootState) => state.userState.user);
-   const currentRecipient = notification.recipients.find(
-     (recipient) => recipient.taxId === currentUser.fiscal_number
-   );
+   * Determination of the recipient that corresponds to the person
+   * whose notifications are displayed, which can be either the logged user
+   * or a different person who has given delegation powers to her.
    */
 
+  // must obtain data from Redux store outside conditionals, even when only one of these two
+  // items will be needed
+  const currentUser = useAppSelector((state: RootState) => state.userState.user);
+  const delegatorsFromStore = useAppSelector((state: RootState) => state.generalInfoState.delegators);
+
+  // I defined functions since linter rules disable the definition of lets, and there is a complex logic 
+  // for the delegator case which I prefer not to code inside a ternary expression
+  function fiscalNumberDaDelegator() {
+    const currentDelegatorFromStore = delegatorsFromStore ? delegatorsFromStore.find(delegatorFromStore => delegatorFromStore.mandateId === mandateId) : null;
+    return currentDelegatorFromStore ? currentDelegatorFromStore.delegator?.fiscalCode : null;
+  }
+
+  function fiscalNumberDaUser() {
+    return currentUser.fiscal_number;
+  }
+
+  // finally the determination for the recipient
+  const fiscalNumberForNotification = mandateId ? fiscalNumberDaDelegator() : fiscalNumberDaUser();
```

Comment:
@clombardi l'algoritmo e la logica sono giusti. proverei a scrivere un'unica function con il nome getCurrentFiscalCode che accorpi tutto l'algoritmo e restituisca direttamente il codice fiscale "definitvo". In questo modo potremo spostare questa function in un service o utility separato e testarne la logica con unit test dedicato.